File: archive/3.2/src/net/geocentral/geometria/action/GSelectAllAction.java
/**
 * Copyright 2000-2013 Geometria Contributors
 * http://geocentral.net/geometria
 * 
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License
 * http://www.gnu.org/licenses
 */
package net.geocentral.geometria.action;

import java.util.Set;

import net.geocentral.geometria.model.GDocument;
import net.geocentral.geometria.model.GFigure;
import net.geocentral.geometria.model.GSelectable;
import net.geocentral.geometria.model.GSolid;

import org.apache.log4j.Logger;

public class GSelectAllAction implements GAction {

    private static Logger logger = Logger.getLogger("net.geocentral.geometria");

    public boolean execute() {
        logger.info("");
        GDocumentHandler documentHandler = GDocumentHandler.getInstance();
        GDocument document = documentHandler.getActiveDocument();
        GFigure figure = document.getSelectedFigure();
        GSolid solid = figure.getSolid();
        solid.selectAll();
        figure.repaint();
        Set<GSelectable> selection = solid.getSelection();
        logger.info(selection);
        return true;
    }
}

File: js/geometria/action/GSelectAllAction.js
/**
 * Copyright (C) 2000-2014 Geometria Contributors
 * http://geocentral.net/geometria
 * 
 * Geometria is free software released under the MIT License
 * http://opensource.org/licenses/MIT
 */
define([
    "geometria/GDictionary",
    "geometria/GFiguresContainer",
    "geometria/GUtils"
], function(dict, figuresContainer, utils) {

    return {

        figureSpecific: true,

        enableAtPlayBack: true,

        label: dict.get("action.SelectAll"),
        
        execute: function() {
            var figure = figuresContainer.getSelectedFigure();
            figure.selectAll();
            return {};
        }
    };
});